plugins/storage/volume/default/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/CloudStackPrimaryDataStoreLifeCycleImpl.java

```diff
@@ -133,7 +133,7 @@ public DataStore initialize(Map<String, Object> dsInfos) {
         String providerName = (String)dsInfos.get("providerName");
         HypervisorType hypervisorType = HypervisorType.None;
         if(dsInfos.containsKey("hypervisorType")) {
-            hypervisorType = (HypervisorType)dsInfos.get("hypervisorType");
+            hypervisorType = HypervisorType.getType((String)dsInfos.get("hypervisorType"));
```

Comment:
@alexandru-bagu in your latest commit, hypervisor type is casted to string again

Reply:
@weizhouapache The issue I created this PR for is that dsInfos.get("hypervisorType") is sometimes String or sometimes missing. As I highlighted in my previous post, both String and HypervisorType is used so I am not sure about how to proceed. We could check types but I don't believe that is correct.